website/docs/guides/upgrade.html.markdown

```diff
@@ -0,0 +1,158 @@
+```
+
+### Running the upgrade
+
+In order to configure and execute upgrade of NSXT edges, hosts, and managers, use `nsxt_upgrade_run` resource. 
+
+This resource will start with configuration of upgrade unit groups and upgrade plan settings for `EDGE`and `HOST` 
+components. Host component groups may be created within this resource with custom host lists and attributes, or it can 
+use the predefined groups which NSX provides.
+
+Following the configuration of the edge, host groups, it will execute the upgrade for each component. If there are 
+either `EDGE` or `HOST` groups disabled from the upgrade, the corresponding component will be paused after enabled 
+groups get upgraded. At the same time, `MP` upgrade won't be processed, because that requires `EDGE` and `HOST` to be 
+fully upgraded. Refer to the exposed attribute for current upgrade state details. For example, check `upgrade_plan`
+for current upgrade plan, which also includes the plan not specified in this resource and `state` for upgrade status of 
+each component and UpgradeUnitGroups in the component. For more details, please check NSX admin guide.
+
+In the example below, host_group `HostWithDifferentSettings` will use different settings than the defaulted ones. The 
+group configuration is initiated with the `nsxt_upgrade_run` while the other groups in this example are predefined and 
+consumed using the data sources `nsxt_edge_upgrade_group` and `nsxt_host_upgrade_group`.
+
+```hcl
+resource "nsxt_upgrade_run" "run" {
+  upgrade_prepare_ready_id = data.nsxt_upgrade_prepare_ready.ready.id
+
+  edge_group {
+    id      = data.nsxt_edge_upgrade_group.eg1.id
+    enabled = false
+  }
+
+  edge_group {
+    id      = data.nsxt_edge_upgrade_group.eg2.id
+    enabled = true
+  }
+
+  host_group {
+    id       = data.nsxt_host_upgrade_group.hg1.id
+    parallel = true
+  }
+  host_group {
+    display_name = "HostWithDifferentSettings"
+    parallel     = false
+    hosts        = ["375c260d-3196-487c-b5eb-046de958f6af"]
+  }
+
+  edge_upgrade_setting {
+    parallel           = true
+    post_upgrade_check = true
+  }
+
+  host_upgrade_setting {
+    parallel           = true
+    post_upgrade_check = true
+  }
+
+  host_group {
+    display_name = "TEST123"
+    parallel     = false
+    hosts        = ["2fa96cdc-6b82-4284-a69a-18a21a6b6d0c"]
+  }
+}
+```
+
+### Post upgrade checks
+
+Upgrade post check data sources can be used to examine the results of the edge and host upgrades, to cunclude if the 
+upgrade has completed successfully.
+
+The example below uses Terraform `check` to test the upgrade results.
+
+```hcl
+data "nsxt_upgrade_postcheck" "pc1" {
+  upgrade_run_id = nsxt_upgrade_run.run.id
+  type           = "EDGE"
+}
+
+data "nsxt_upgrade_postcheck" "pc2" {
+  upgrade_run_id = nsxt_upgrade_run.run.id
+  type           = "HOST"
+}
+
+check "edge_post_check" {
+  assert {
+    condition     = data.nsxt_upgrade_postcheck.pc1.failed_group == null
+    error_message = "test"
```

Comment:
super pedant nit: can we have a better error message - both above and at line 155.